### DISCUSSION TITLE
Is TOM TOM street network data is required if I want to integrate historical and live traffic into Valhalla routing engine

### DISCUSSION THREAD

**Discussion by NitinMehra0019** [valhalla/valhalla#4564] [Q&A]
Hi there,
I was going through  https://gis-ops.com/traffic-in-valhalla/ guide and it is written here that if I want to integrate live or historical traffic data into routing engine then I need to build the routing graph first from TOM TOM MultiNet street network data. 

Can someone explain me why it is required to have same base network to work with the traffic data.

![Screenshot from 2024-02-06 10-29-56](https://github.com/valhalla/valhalla/assets/84515402/a44b09c3-3f29-4420-b367-f28cc6044420)

**Comment by Trietes** [comment]
It is not required. But if you have traffic data from e.g. TomTom and you would like to route on an OSM map you have to match the traffic data to the map material. You have to see which traffic data you have and how you can bring it onto the valhalla graph. And you have to check the licences of your traffic source. They might not allow matching the traffic data to other map materials.

In the docs its explained how you can map OSM traffic data onto the valhalla graph (https://valhalla.github.io/valhalla/mjolnir/historical_traffic/). If you have any other format you have to see how you can do it.

**Reply by nilsnolde** [reply]
> It is not required. But if you have traffic data from e.g. TomTom and you would like to route on an OSM map you have to match the traffic data to the map material.

Actually that won't be possible with TomTom (historical) data, there's no geometry information in those files. So for TomTom (haven't seen HERE traffic data yet but very surely the same..) you do need the base map as well. The OP's screenshot is from an article I wrote, feel free to ask for commercial support on enquiry@gis-ops.com @NitinMehra0019, we can provide the conversion as a service or you can even license [our tool](https://github.com/gis-ops/prop2osm-public) to do so on your own machine.
